HPCC-17130 Mine button to return owner results

miguelvazq opened the pull request:
Add a owner filter checkbox called "Mine" that will allow that user to
return the currently logged on user's items for specific:
- Workunits
- Files
- DFU Workunits
- Published Queries

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [x] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

miguelvazq commented:
@GordonSmith Please review.

HPCCSmoketest commented:
Automated Smoketest: :x:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: b68b8701e51d217741adb2ed3392a691221e2da8
Build: success
ECL Watch:  Rebuilding Site

| lint errors | 
|---|
|5 |



Lint error(s): 
eclwatch/DFUQueryWidget.js: line 457, col 18, Unnecessary semicolon.
eclwatch/GetDFUWorkunitsWidget.js: line 271, col 18, Unnecessary semicolon.
eclwatch/GetDFUWorkunitsWidget.js: line 267, col 13, WsAccount is not defined.
eclwatch/QuerySetQueryWidget.js: line 180, col 18, Unnecessary semicolon.
eclwatch/WUQueryWidget.js: line 339, col 18, Unnecessary semicolon.

Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |765 |765 |0 |
|test (thor) |679 |679 |0 |
|test (roxie) |792 |792 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   36 sec (00:00:36) |  183 sec (00:03:03) |  127 sec (00:02:07) |    7 sec (00:00:07) |   38 sec (00:00:38) | 1141 sec (00:19:01) |   32 sec (00:00:32) | 1564 sec (00:26:04) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: f9f7cc750dafa5900a9ffffabf0f55152018a2b2
Build: success
ECL Watch:  Rebuilding Site
Rebuild: success

Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |83 |83 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |765 |765 |0 |
|test (thor) |679 |679 |0 |
|test (roxie) |792 |792 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   16 sec (00:00:16) |  180 sec (00:03:00) |  126 sec (00:02:06) |    7 sec (00:00:07) |   38 sec (00:00:38) | 1135 sec (00:18:55) |   35 sec (00:00:35) | 1537 sec (00:25:37) |

miguelvazq commented:
@GordonSmith Rebased.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: f1c67e97b9c1816b30170bab1c55029509a03185
Build: success
ECL Watch:  Rebuilding Site
Rebuild: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |85 |85 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   40 sec (00:00:40) |  182 sec (00:03:02) |  128 sec (00:02:08) |    6 sec (00:00:06) |   22 sec (00:00:22) | 1135 sec (00:18:55) |   16 sec (00:00:16) | 1529 sec (00:25:29) |

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 5eee60188259a6c091cc5603ba7a1e28c3ecaacd
Build: success
ECL Watch:  Rebuilding Site
Rebuild: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |85 |85 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |759 |759 |0 |
|test (thor) |686 |686 |0 |
|test (roxie) |799 |799 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   10 sec (00:00:10) |  182 sec (00:03:02) |  113 sec (00:01:53) |    7 sec (00:00:07) |   23 sec (00:00:23) | 1130 sec (00:18:50) |   16 sec (00:00:16) | 1481 sec (00:24:41) |

GordonSmith commented on esp/src/eclwatch/DFUQueryWidget.js:
```diff
@@ -449,6 +462,27 @@ define([
+                if (lang.exists("MyAccountResponse.username", response)) {
+                    context.userName = response.MyAccountResponse.username;
+                }
+            });
```
I think you can get the global user id from dojoConfig?

GordonSmith reviewed (approved):
Looks ok - it may be better to use the global userID from dojoConfig?

richardkchapman commented:
@miguelvazq

miguelvazq commented:
@tonymkirk Any feedback on mine button placement? Does it make more sense to have it globally and affect all components vs. individual list views?

tonymkirk commented:
I could see a global "mine" switch being useful, perhaps, but not in lieu of individual list views.

richardkchapman commented:
@miguelvazq Needs rebasing for a clean merge

miguelvazq commented:
Will rebase and move the username to a dojoConfig based call.

Commit:
HPCC-17130 Mine button to return owner results

Add a owner filter checkbox called "Mine" that will allow that user to
return the currently logged on user's items for specific:

- Workunits
- Files
- DFU Workunits
- Published Queries

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

miguelvazq commented:
@GordonSmith Rebased and added dojoConfig

GordonSmith reviewed (approved):
Looks good.
